OrbitLinuxTracing/PerfEventRingBuffer.h

```diff
@@ -30,7 +31,13 @@ class PerfEventRingBuffer {
   bool HasNewData();
   void ReadHeader(perf_event_header* header);
   void SkipRecord(const perf_event_header& header);
-  void ConsumeRecord(const perf_event_header& header, void* record);
+
+  template <typename T>
+  void ConsumeRecord(const perf_event_header& header, T* record) {
+    CHECK(header.size == sizeof(T));
+    ConsumeRecord(header, record, sizeof(T));
+  }
+  void ConsumeRecord(const perf_event_header& header, void* record, size_t size);
```

Comment:
This should be private. If you need to read from the ring buffer without reading the whole record you are not really "consuming the record", you should use `ReadValueAtOffset` and `SkipRecord` as it's already done in many other places.

Reply:
done.